libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/BotFrameworkClientImpl.java

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.connector.authentication;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.bot.connector.Async;
+import com.microsoft.bot.connector.skills.BotFrameworkClient;
+import com.microsoft.bot.restclient.serializer.JacksonAdapter;
+import com.microsoft.bot.schema.Activity;
+import com.microsoft.bot.schema.ChannelAccount;
+import com.microsoft.bot.schema.ConversationAccount;
+import com.microsoft.bot.schema.ConversationReference;
+import com.microsoft.bot.schema.RoleTypes;
+import com.microsoft.bot.schema.Serialization;
+import com.microsoft.bot.schema.TypedInvokeResponse;
+import okhttp3.HttpUrl;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.concurrent.CompletableFuture;
+
+public class BotFrameworkClientImpl extends BotFrameworkClient {
+    private final ServiceClientCredentialsFactory credentialsFactory;
+    private OkHttpClient httpClient;
```

Comment:
This httpClient is not being assigned to right now. It's supposed to be assigned through [constructor](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Connector/Authentication/BotFrameworkClientImpl.cs#L28) but then we should modify the call in [PBFA](https://github.com/southworks/botbuilder-java/blob/internal/feature/southworks/cloudadapter/base/libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ParameterizedBotFrameworkAuthentication.java#L176). Maybe we can see this when we put the whole thing together

Reply:
I fixed this generating the httpClient in the constructor, not necessary to passing by parameter. I followed the `BotFrameworkHttpClient` approach here